app/controllers/posts_controller.rb

```diff
@@ -16,11 +16,12 @@ def new
   def create
     new_post = Post.new(post_params)
     current_user.posts << new_post
-
+    new_post.update_post_counter
     if new_post.save
-      redirect_to user_posts_path(new_post.user.id), notice: 'Post created!'
+      flash[:notice] = 'Post created!'
+      redirect_to user_posts_path(new_post.user.id)
     else
-      flash.now[:alert] = 'Failed to publish post!'
+      flash.now[:alert] = 'Failed to create post!'
       render :new
     end
   end
```

Comment:
- [ ] I highlighted this method because is where you did some change but you should implement N+1 in your `index` method about fetching all posts and their comments for a user.
instead of `@posts = @user.top_most_recent_posts` you will `include` comments that are related to the posts

Reply:
Hi @DeliceLydia
When I tried to include comments the n+1 problems is raised but without including the comments it working fine

Reply:
Hi @DeliceLydia 
Here is a screenshot when I include comments
![Screen Shot 2021-11-02 at 22 30 40](https://user-images.githubusercontent.com/55432600/139954685-b16a85a2-14e5-4d05-ae61-68239747b6fd.png)

Reply:
Hi @DeliceLydia 
Could you please help me with a screenshot of what eager loading recommendation you are facing.